### DISCUSSION TITLE
MeshCentral in a custom React Application

### DISCUSSION THREAD

**Discussion by sufailDesk** [Ylianst/MeshCentral#4417] [Q&A]
Hello, Team Meshcentral
Is there a more effective method or examples of projects to use when using MeshCentral to build specialised React applications?

**Comment by Ylianst** [comment]
I have never built a React application myself, but MeshCentral is websocket based. You can add `?trace=1` to the URL when logged into MeshCentral and open the browser's console to see all of the messages. It's JSON messages going both ways, very easy to encode and parse. Once you understand this, you should be able to create a React application that opens the same websocket an do all the same operations. Hope that helps.
